server/sonar-server/src/main/java/org/sonar/server/db/migrations/PlatformDatabaseMigration.java

```diff
@@ -64,15 +68,34 @@ public PlatformDatabaseMigration(RubyBridge rubyBridge, PlatformDatabaseMigratio
 
   @Override
   public void startIt() {
```

Comment:
Not sure to understand the implementation of startIt(). Why don't you do something like the following ?

```
// no condition on lock.isLocked()
if (lock.tryLock()) {
  try {
    startAsynchronousDBMigration();
  } finally {
    lock.unlock();
  }
}
```